src/main/scala/org/ergoplatform/mining/mining.scala

```diff
@@ -0,0 +1,32 @@
+package org.ergoplatform
+
+import org.bouncycastle.math.ec.ECPoint
+import scapi.sigma.BcDlogFp
+import scapi.sigma.DLogProtocol.DLogProverInput
+import sigmastate.interpreter.CryptoConstants
+import sigmastate.interpreter.CryptoConstants.EcPointType
+
+package object mining {
+
+  type PrivateKey = BigInt
+
+  val PublicKeyLength: Byte = 33
+
+  val group: BcDlogFp[EcPointType] = CryptoConstants.dlogGroup
+
+  val q: BigInt = group.q
+
+  private val hashFn: NumericHash = new NumericHash(q)
```

Comment:
Blake2b512 is not needed, in case 25519 q ~= 252 bits, so you can just set four most significant bits of 256-bits hash to zero.

Reply:
replaced underlying hash to Blake2b256.